WebRoot/js/zimbraMail/calendar/controller/ZmCalendarTreeController.js

```diff
@@ -538,10 +538,25 @@ function() {
 */
 ZmCalendarTreeController.prototype._addExternalCalendarListener =
 function() {
-	var dialog = this.getExternalCalendarDialog();
-    dialog.popup();
+	return AjxRpc.invoke(null, "/service/extension/oauth2/info/google", null, new AjxCallback(this, this._handleGoogleConfig), true);
 };
 
+ZmCalendarTreeController.prototype._handleGoogleConfig =
+function(response){
+	try {
+		var responseJson = JSON.parse(response.text);
+		let isGoogleAuthConfigured = false;
+
+		if(responseJson.data && responseJson.data.client_id){
+			isGoogleAuthConfigured = true;
+		}
+
+		var dialog = this.getExternalCalendarDialog(isGoogleAuthConfigured);
+		dialog.popup();
+	} catch(e) {
```

Comment:
we don't need try/catch for all statements, just JSON.parse

Reply:
ok